CRM/Mailing/BAO/Mailing.php

```diff
@@ -218,30 +219,31 @@ public static function getRecipients($mailingID) {
       $location_filter,
       "$email.email IS NOT NULL",
       "$email.email != ''",
+      "mg.mailing_id = #mailingID",
       'temp.contact_id IS NULL',
     );
 
     if ($isSMSmode) {
-      $phone = CRM_Core_DAO_Phone::getTableName();
+      $includeFilters = array(
+        "mg.group_type = 'Include'",
+        'mg.search_id IS NULL',
+        "gc.status = 'Added'",
+        "$contact.do_not_sms",
+        "$contact.is_opt_out = 0",
+        "$contact.is_deceased <> 1",
+        "$phone.phone_type_id = " . CRM_Utils_Array::value('Mobile', CRM_Core_OptionGroup::values('phone_type', TRUE, FALSE, FALSE, NULL, 'name')),
```

Comment:
does CRM_Core_PseudoConstant::getKey not work here? It's certainly shorter & that OptionGroup::values function is a bit evil